### PR TITLE
mon: fix "fs new" pool metadata update, tests

### PR DESCRIPTION
**Pull request by gregsfortytwo**

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
So far I just turned on a vstart cluster. I have no confidence at all in my qa suite python update.

**Comment by gregsfortytwo** [comment]
retest this please

File: src/mon/FSCommands.cc [hunk 1]
```diff
@@ -193,13 +188,20 @@ class FsNewHandler : public FileSystemCommandHandler
...
     }
-
+    
+    // if we're running as luminous, we have to set the pool application metadata
     if (mon->osdmon()->osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS ||
 	mon->osdmon()->pending_inc.new_require_osd_release >= CEPH_RELEASE_LUMINOUS) {
+      if (!mon->osdmon()->is_writeable()) {
+	// not allowed to write yet, so retry when we can
+	mon->osdmon()->wait_for_writeable(op, new PaxosService::C_RetryMessage(mon->mdsmon(), op));
+	return EAGAIN;
```
Comment (liewegas) [inline src/mon/FSCommands.cc]: -EAGAIN

File: qa/tasks/cephfs/filesystem.py [hunk 2]
```diff
@@ -465,6 +467,17 @@ def create(self):
...
         self.getinfo(refresh = True)
 
+        
+    def check_pool_application(self, pool_name):
+        osd_map = self.mon_manager.get_osd_dump_json()
+        for pool in osd_map['pools']:
+            if pool['pool_name'] == pool_name:
+                if "application_metadata" in pool:
+                    if not "cephfs" in pool['application_metadata']:
+                        raise RuntimeError("Pool %p does not name cephfs as application!".\
```
Comment (dillaman) [inline qa/tasks/cephfs/filesystem.py]: Does this need to be guarded against rados test suites where the luminous "min osd release" upgrade is performed at the end of the test (after fs new)?

Comment (gregsfortytwo) [inline qa/tasks/cephfs/filesystem.py]: It already is with the ```if "application_metadata"``` check?

Comment (liewegas) [inline qa/tasks/cephfs/filesystem.py]: Yeah I think this is right as is

**Comment by gregsfortytwo** [comment]
Testing Sage's EAGAIN-fixed ci build against the updated suite from above and it passed one fs test (http://pulpito.ceph.com/gregf-2017-08-10_18:16:59-fs-wip-20891-pool-metadata-distro-basic-smithi/), and going through the logs we're getting proper "application_metadata": "cephfs" output on the osd dump for the first time I've seen. Plus that means the python works now.

Have a kraken upgrade test running at http://pulpito.ceph.com/gregf-2017-08-10_19:55:49-upgrade:kraken-x:parallel-wip-20891-pool-metadata-distro-basic-smithi/

**Comment by gregsfortytwo** [comment]
@liewegas the upgrade suite passed 11/12 and failed on a Bluestore fsck crash.

**Pull request merged**